src/Laravel/ApiPlatformProvider.php

```diff
@@ -1130,44 +1129,78 @@ private function registerGraphQl(Application $app): void
+                $app->make(GraphQlReadProvider::class),
+                new ServiceLocator($resolvers),
+            );
+        });
+
+        $app->alias(ResolverProvider::class, 'api_platform.graphql.state_provider.resolver');
+
+        $app->singleton(GraphQlDenormalizeProvider::class, function (Application $app) {
+            return new GraphQlDenormalizeProvider(
+                $this->app->make(ResolverProvider::class),
+                $app->make(SerializerInterface::class),
+                $app->make(GraphQlSerializerContextBuilder::class)
+            );
+        });
+
+        $app->alias(GraphQlDenormalizeProvider::class, 'api_platform.graphql.state_provider.denormalize');
+
+        $app->singleton('api_platform.graphql.state_provider.parameter', function (Application $app) {
+            $tagged = iterator_to_array($app->tagged(ParameterProviderInterface::class));
+            $tagged['api_platform.serializer.filter_parameter_provider'] = $app->make(SerializerFilterParameterProvider::class);
+
+            return new ParameterProvider(
+                new ParameterValidatorProvider(
+                    new SecurityParameterProvider(
+                        $app->make(GraphQlDenormalizeProvider::class),
+                        $app->make(ResourceAccessCheckerInterface::class)
+                    ),
+                ),
+                new ServiceLocator($tagged)
+            );
+        });
+
+        $app->singleton('api_platform.graphql.state_provider.access_checker', function (Application $app) {
+            return new AccessCheckerProvider($app->make('api_platform.graphql.state_provider.parameter'), $app->make(ResourceAccessCheckerInterface::class));
+        });
+
+        $app->singleton(NormalizeProcessor::class, function (Application $app) {
+            return new NormalizeProcessor(
+                $app->make(SerializerInterface::class),
+                $app->make(GraphQlSerializerContextBuilder::class),
+                $app->make(Pagination::class)
+            );
+        });
+        $app->alias(NormalizeProcessor::class, 'api_platform.graphql.state_processor.normalize');
 
         $app->singleton('api_platform.graphql.state_processor', function (Application $app) {
             return new WriteProcessor(
-                new NormalizeProcessor(
-                    $app->make(SerializerInterface::class),
-                    $app->make(GraphQlSerializerContextBuilder::class),
-                    $app->make(Pagination::class)
-                ),
+                $app->make('api_platform.graphql.state_processor.normalize'),
                 $app->make(CallableProcessor::class),
             );
         });
 
         $app->singleton(ResolverFactoryInterface::class, function (Application $app) {
             return new ResolverFactory(
-                $app->make('api_platform.graphql.state_provider'),
+                $app->make('api_platform.graphql.state_provider.access_checker'),
```

Comment:
@vinceAmstoutz I splitted the system providers for Laravel!

Reply:
@soyuka :tada: Can we prepare documentation now and merge it when your PR will be merged?